### ISSUE TITLE
OSM destroys cluster if enabled for a pod with hostNetwork

### ISSUE THREAD

**Issue by plwhite** [openservicemesh/osm#4359]
**Bug description**:

**Affected area (please mark with X where applicable)**:
- Install                    [ ]
- SMI Traffic Access Policy  [ ]
- SMI Traffic Specs Policy   [ ]
- SMI Traffic Split Policy   [ ]
- Permissive Traffic Policy  [ ]
- Ingress                    [ ]
- Egress                     [ ]
- Envoy Control Plane        [ ]
- CLI Tool                   [ ]
- Metrics                    [ ]
- Certificate Management     [ ]
- Sidecar Injection          [X]
- Logging                    [ ]
- Debugging                  [ ]
- Tests                      [ ]
- Demo                       [ ]
- CI System                  [ ]

**Expected behavior**:

Cluster is not destroyed

**Steps to reproduce the bug (as precisely as possible)**:

Create a pod with `hostNetwork` set true in an OSM managed namespace. OSM inserts an init container (through the mutating admissions controller), which sets up iptables rules in the container. Since the pod is in the node network namespace the resulting iptables rules affect the node as a whole, and things break dramatically.

**How was OSM installed?**:

~~~
wget https://github.com/openservicemesh/osm/releases/download/v0.9.2/osm-v0.9.2-linux-amd64.tar.gz
tar zxvf osm-v0.9.2-linux-amd64.tar.gz
sudo mv linux-amd64/osm /usr/local/bin
sudo chmod +x /usr/local/bin/osm

osm install
osm namespace add blah
~~~

**Anything else we need to know?**:

Other service meshes exclude any pod with `hostNetwork` set to true - see for example https://github.com/linkerd/linkerd2/issues/366 and https://istio.io/latest/docs/ops/common-problems/injection/

There are obvious workarounds here by explicitly annotating pods so OSM doesn't do this, but having a "take down my entire cluster" capability (which is the default for `hostNetwork` pods) is suboptimal.

**Bug report archive**:

**Environment**:
- OSM version (use `osm version`): `Version: v0.9.2; Commit: 7df9badece2f251515a2bc5e5ac79438b0c58812; Date: 2021-08-19-21:02`
- Kubernetes version (use `kubectl version`): `Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.20", GitCommit:"1f3e19b7beb1cc0110255668c4238ed63dadb7ad", GitTreeState:"clean", BuildDate:"2021-06-16T12:51:17Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}`
- Size of cluster (number of worker nodes in the cluster): 1
- Others:

**Comment by shashankram** [comment]
Resolved by https://github.com/openservicemesh/osm-docs/pull/283. If `hostNetwork: true` is set on the pod spec, it will not be injected with a sidecar.

**Comment by plwhite** [comment]
Thanks - really pleased to see this fixed so quickly!

**Comment by shashankram** [comment]
> Thanks - really pleased to see this fixed so quickly!

Hey @plwhite, you are welcome! Thanks for reporting this critical issue and helping us improve. The fix should be available in release v1.0 (due in the next few weeks).
